### PR TITLE
PR: Fix recursive debugger

### PR DESCRIPTION
**Pull request by impact27**
Addresses https://github.com/spyder-ide/spyder/issues/14472

### CODE REVIEW

File: spyder_kernels/customize/spydercustomize.py [hunk 1]
```diff
@@ -583,6 +571,54 @@ def runfile(filename=None, args=None, wdir=None, namespace=None,
 builtins.runfile = runfile
 
 
+def enter_debugger(filename, continue_if_has_breakpoints, code_format):
```
Comment (ccordoba12) [inline spyder_kernels/customize/spydercustomize.py:574]: Given that this function is quite long now, I'd prefer to move it to `spyder_kernels/customize/spyderpdb.py`

File: spyder_kernels/customize/spydercustomize.py [hunk 2]
```diff
@@ -583,6 +571,54 @@ def runfile(filename=None, args=None, wdir=None, namespace=None,
...
+    debugger._user_requested_quit = False
+
+    if os.name == 'nt':
+        filename = filename.replace('\\', '/')
+
+    code = code_format.format(repr(filename))
+
+    if recursive:
+        # Enter recursive debugger
+        parent_debugger.message("ENTERING RECURSIVE DEBUGGER")
```
Comment (ccordoba12) [inline spyder_kernels/customize/spydercustomize.py:610]: Are you sure you want to use all caps for this message? I mean, is it really such an important message for that?

File: spyder_kernels/customize/spydercustomize.py [hunk 3]
```diff
@@ -583,6 +571,54 @@ def runfile(filename=None, args=None, wdir=None, namespace=None,
...
+        sys.call_tracing(debugger.run, (code, globals, locals))
+        parent_debugger.message("LEAVING RECURSIVE DEBUGGER")
```
Comment (ccordoba12) [inline spyder_kernels/customize/spydercustomize.py:612]: The same applies here, of course.

**Review by ccordoba12 — COMMENTED** [review]
Thanks @impact27 for your help with this! I left some small comments for you.

File: spyder_kernels/customize/spydercustomize.py (hunk 2 above)
Comment (impact27) [reply spyder_kernels/customize/spydercustomize.py:610]: This is to be consistant with https://github.com/ipython/ipython/blob/106a841993ee773fbe4045b8aec5cb37d6cc3eaa/IPython/core/debugger.py#L606, But I think we can drop the message.

**Review by ccordoba12 — APPROVED** [review]
Looks good to me now, thanks @impact27!

**Pull request merged**